### ISSUE TITLE
Set exitsed Column/RowDefinition's SharedSizeGroup not update layout

### ISSUE THREAD

**Issue by Flithor** [AvaloniaUI/Avalonia#12300]
**Describe the bug**

In my custom control([axaml](https://github.com/Flithor/ReusableCodes/blob/main/AvaloniaUI/FormField.axaml)|[code](https://github.com/Flithor/ReusableCodes/blob/main/AvaloniaUI/FormField.axaml.cs))
If not contains belows code lines, the share size will not work.
https://github.com/Flithor/ReusableCodes/blob/023caf4918e03fb2bd0c067df6b858450ddfbffd/AvaloniaUI/FormField.axaml.cs#L100-L102

https://github.com/Flithor/ReusableCodes/blob/023caf4918e03fb2bd0c067df6b858450ddfbffd/AvaloniaUI/FormField.axaml.cs#L122-L124

**To Reproduce**
Use my control code and try it.

**Expected behavior**
I think when I set the `SharedSizeGroup` of Definition, it should refresh the layout.

**Screenshots**
Reseted the defination collection in code:
![image](https://github.com/AvaloniaUI/Avalonia/assets/23412916/386abf71-36b2-4eec-b005-13665465f021)
Not reseted:
![image](https://github.com/AvaloniaUI/Avalonia/assets/23412916/8db23032-41aa-46f6-8f91-f5e507bf2430)

**Desktop**

- OS: Windows 11
- Version 0.10.21

**Comment by timunie** [comment]
@Flithor please double-check if it works using 11.0. Thx 🙏

**Comment by Flithor** [comment]
> @Flithor please double-check if it works using 11.0. Thx 🙏

My work is on 0.10.21, and may never update to 0.11..

**Comment by timunie** [comment]
Yeah, you would need to write a small test app to verify. However, the thing is that 0.10.x will not get much bug fixes anymore, esp. if there is a somewhat working workaround (unless you have paid support, which I don't know). But for 11.0, a fix is needed if bug is still present. 

Thanks for your understanding.

**Comment by timunie** [comment]
Tested in 11.0 and seems to work as expected. @avaloniaui-team please decide if this should be closed as not planned or not.
